tests/suite/nginxgateway_test.go

```diff
@@ -0,0 +1,180 @@
+				ngfPodName, err := getNGFPodName()
+				Expect(err).ToNot(HaveOccurred())
+
+				_ = verifyAndReturnNginxGateway(nginxGatewayNsname)
+
+				logs, err := resourceManager.GetPodLogs(ngfNamespace, ngfPodName, &core.PodLogOptions{
+					Container: "nginx-gateway",
+				})
+				Expect(err).ToNot(HaveOccurred())
+
+				Expect(logs).To(ContainSubstring("\"level\":\"debug\""))
+			})
+		})
+
+		When("default log level is used", func() {
+			It("only outputs info logs and the status is accepted and true", func() {
+				teardown(releaseName)
+
+				cfg := getDefaultSetupCfg()
+				cfg.infoLogLevel = true
+				setup(cfg)
+
+				ngfPodName, err := getNGFPodName()
+				Expect(err).ToNot(HaveOccurred())
+
+				nginxGateway := verifyAndReturnNginxGateway(nginxGatewayNsname)
+				Expect(nginxGateway.Status.Conditions[0].ObservedGeneration).To(Equal(int64(1)))
+
+				logs, err := resourceManager.GetPodLogs(ngfNamespace, ngfPodName, &core.PodLogOptions{
+					Container: "nginx-gateway",
+				})
+				Expect(err).ToNot(HaveOccurred())
+
+				Expect(logs).ToNot(ContainSubstring("\"level\":\"debug\""))
+			})
+		})
+	})
+
+	When("testing on an existing NGF instance", Ordered, func() {
+		BeforeAll(func() {
+			var err error
+			ngfPodName, err = getNGFPodName()
+			Expect(err).ToNot(HaveOccurred())
+		})
+
+		When("NginxGateway is updated", func() {
+			It("captures the change, the status is accepted and true,"+
+				" and the observed generation is incremented", func() {
+				// previous test has left the log level at info, this test will change the log level to debug
+
+				var observedGeneration int64
+
+				nginxGateway := verifyAndReturnNginxGateway(nginxGatewayNsname)
+				observedGeneration = nginxGateway.Status.Conditions[0].ObservedGeneration
+
+				logs, err := resourceManager.GetPodLogs(ngfNamespace, ngfPodName, &core.PodLogOptions{
+					Container: "nginx-gateway",
+				})
+				Expect(err).ToNot(HaveOccurred())
+
+				Expect(logs).ToNot(ContainSubstring("\"level\":\"debug\""))
+
+				Expect(resourceManager.ApplyFromFiles(files, namespace)).To(Succeed())
+				// need to wait until files are applied, no current function because this is a nginx-gateway crd
+				time.Sleep(2 * time.Second)
+
+				nginxGateway = verifyAndReturnNginxGateway(nginxGatewayNsname)
+				Expect(nginxGateway.Status.Conditions[0].ObservedGeneration).To(Equal(observedGeneration + 1))
+
+				logs, err = resourceManager.GetPodLogs(ngfNamespace, ngfPodName, &core.PodLogOptions{
+					Container: "nginx-gateway",
+				})
+				Expect(err).ToNot(HaveOccurred())
+
+				Expect(logs).To(ContainSubstring(
+					"\"current\":\"debug\",\"msg\":\"Log level changed\",\"prev\":\"info\"",
+				))
+			})
+		})
+
+		When("NginxGateway is deleted", func() {
+			It("captures the deletion and default values are used", func() {
+				Expect(resourceManager.DeleteFromFiles(files, namespace)).To(Succeed())
+				time.Sleep(2 * time.Second) // need to wait until deletion is fully processed
```

Comment:
Instead of sleeping, we should wait until a GET on the NginxGateway resource returns a not found error.